### DISCUSSION TITLE
Errors in parallel calculation and Radiance running

### DISCUSSION THREAD

**Discussion by gitguox** [Eddy3D-Dev/Discussions#3] [Q&A] [answered]
Hello Eddy3d community,

I am new to Eddy3d and am currently working on an outdoor UTCI simulation. I’ve encountered two persistent issues while running Eddy3d for testing, and I’m hoping someone here might be able to assist me. Thanks for anyone who takes the time to look into my problem. Here are the details of the issues I’m facing:

#1. Multi-core parallel computation error: I have followed the instructions on the forum (https://www.cfd-online.com/Forums/openfoam-installation/200437-bluecfd-core-2016-user-compiled-solvers-not-running-parallel.html#post687582), copied the msmpi.dll, and performed computations using multiple cores (set CPUs to -1) which seemed to run normally (from the “Simulation” and “Residuals” components, it appears to be normal). The “Probing” component also reads the wind speed simulation results correctly. However, when using the “Wprobe” component to read the results, I encounter an error. The error message is:
 "_**Solution exception: An element of type EddyLib.Geometry.EddyVector was null; this might be due to contents in a list/array**_"
Is this an issue with the “Wprobe” component itself? Is there any way to read the simulation results from multi-core computations and input them into the “MRT system” component for further calculations?
![1](https://github.com/user-attachments/assets/cc327aca-f6d8-437e-ba75-224c28214bd9)
![residuals](https://github.com/user-attachments/assets/f9595641-f211-4820-8b49-0c9eb6425daf)

#2. Radiance computation error: I have three working computers, and I’ve installed the same versions of Eddy3D, Radiance, and Blue-CFD on all of them, ensuring they were installed in the same manner and path. All computers can compute wind speeds normally with 1 CPU core, and the “Probing” and “Wprobe” component can read the results as expected. However, when I proceed with MRT calculations, the problem arises. Radiance works correctly on one computer, but crashes on the other two. The crash occurs immediately when Radiance reaches the step “_Perform an annual direct-only daylight coefficients simulation…_”, with no UTCI.eddy or MRT.eddy files generated. I’ve noticed that on the two computers where the crash occurs, the “MRT system” component seems to be unable to read the “wind.eddy*” files correctly. Even when I disconnect all inputs at the CFD input end of the “MRT system” component, the same crash occurs. I’ve found a RadiationErrorLog.log in the calculation results folder, which records the error as:
“_**dircalc3 command failed with exit code 1: system - cannot open file ‘Rad\output\dcd_r1.mtx’: No such file or directory
: cannot load matrix**_”
What could be causing this issue?
![2](https://github.com/user-attachments/assets/33e18853-b196-4a2f-af2f-04a04dd356b2)

I have attached the GH file I am using. 
[EDDY-Outdoor UTCI-template.zip](https://github.com/user-attachments/files/17530633/EDDY-Outdoor.UTCI-template.zip)

Current configurations:
Windows 10 & 11
Eddy3d version 0.4.8.0
Rhino 8.8 SR8 (8.8.24163.12481)
Radiance_938fd65c
blueCFD-Core-2020-1-win64-setup

**Comment by kastnerp** [comment]
let me take a look.

**Comment by lemon7up10** [comment]
我和您出现的问题一模一样，请问您解决了吗？

**Reply by gitguox** [reply]
还没解决。。我后来实在搞不定就先忙论文的其它部分去了。我看1月份有更新的版本，我还没来得及测试，不知道这些问题在新版本中是否已经解决了

**Reply by kastnerp** [reply]
I will not be able to help with posts are aren't written in English, sorry!

**Comment by kastnerp** [comment, marked as answer]
Two separate issues here:

**1. Multi‑core / parallel error.** That CFD‑Online thread is about matching BlueCFD's MPI DLL — it's fragile and very version‑specific. The robust fix now is to **switch the run engine to Docker** (right‑click the Run component → Docker): it ships a self‑contained OpenFOAM 12 image and sidesteps the MS‑MPI/BlueCFD DLL matching entirely, and it works the same on Windows and Mac. If you need to stay on BlueCFD, first confirm the case runs **single‑core**, then apply the DLL match from that thread.

**2. Radiance.** You no longer install Radiance by hand. The **Install Engines** component downloads and installs the correct Radiance build automatically (Windows & Mac) into the per‑user Eddy3D folder — run it once and the MRT/UTCI path will find Radiance on its own.

Both of the above are on the current unified **Eddy3D 1.0.1.827** (Rhino Package Manager → `Eddy3D`), so I'd recommend updating first. @lemon7up10 — same recommendation for the parallel error.

Closing as resolved — please open a new thread if the Docker run itself errors (share the console output).

---

*Answer drafted by Claude (Anthropic's AI assistant), at the instruction of Patrick Kastner (Eddy3D maintainer).*
